src/scales/scale.time.js

```diff
@@ -36,10 +40,94 @@ module.exports = function(Chart) {
 			},
 		},
 		ticks: {
-			autoSkip: false
+			autoSkip: false,
+			mode: 'series',   // 'linear|series'
+			source: 'auto'    // 'auto|labels'
 		}
 	};
 
+	function buildLookupTable(ticks, min, max, linear) {
+		var ilen = ticks.length;
+		var table = [];
+		var i, prev, curr, next, decimal;
+
+		if (ilen === 0) {
+			return table;
+		}
+
+		if (ticks[0] !== min) {
+			table.push({time: min, decimal: 0});
+		}
+
+		for (i=0, ilen=ticks.length; i<ilen; ++i) {
+			next = ticks[i+1] || 0;
+			prev = ticks[i-1] || 0;
+			curr = ticks[i];
+
+			// only add points that breaks the scale linearity
+			if (Math.round((next+prev)/2) !== curr) {
+				decimal = linear? (curr - min) / (max - min) : ilen > 1? i/(ilen-1) : 0;
```

Comment:
should probably add a space before the `?` characters. we're also inconsistent about spacing around the `/` characters on this line

I think a set of parens here would help clarify order of operations. how about:

    linear ? (curr - min) / (max - min) : (ilen > 1 ? i / (ilen-1) : 0);

(at least i'm assuming that's the order of operations and it isn't `(linear ? (curr - min) / (max - min) : ilen > 1) ? i / (ilen-1) : 0;`)